sky/provision/oci/query_utils.py

```diff
@@ -472,44 +474,113 @@ def delete_vcn(cls, net_client, skypilot_vcn, skypilot_subnet,
 
     @classmethod
     @debug_enabled(logger)
-    def add_ingress_rules(cls, region: str, ports: Set[int]):
-        if ports is None or len(ports) == 0:
+    def find_or_create_nsg(cls, region: str, cluster_name: str) -> str:
+        net_client = oci_adaptor.get_net_client(
+            region, oci_utils.oci_config.get_profile())
+
+        compartment = cls.find_compartment(region)
+
+        list_vcns_resp = net_client.list_vcns(
+            compartment_id=compartment,
+            display_name=oci_utils.oci_config.VCN_NAME,
+            lifecycle_state='AVAILABLE',
+        )
+
+        if not list_vcns_resp:
+            raise ResourcesUnavailableError('The VCN is not available')
+
+        vcn = list_vcns_resp.data[0]
+
+        list_nsg_resp = net_client.list_network_security_groups(
+            compartment_id=compartment,
+            vcn_id=vcn.id,
+            limit=1,
+            display_name=f'nsg_{cluster_name}',
```

Comment:
Can we have a constant for the nsg name format? It would help us reduce the possibility to only modify some of them in the future.

Reply:
add the nsg name prefix as constant.